src/main/java/nl/hsac/fitnesse/fixture/slim/web/BrowserTest.java

```diff
@@ -2001,13 +2001,43 @@ protected String savePageSource(String fileName, String linkText) {
     }
 
     /**
-     * Takes screenshot from current page
+     * Takes screenshot of page as currently visible
      * @param basename filename (below screenshot base directory).
      * @return location of screenshot.
      */
     public String takeScreenshot(String basename) {
+        return takeScreenshotWith(
+            basename,
+            name ->
+                createScreenshot(name)
+        );
+    }
+
+    /**
+     * Takes screenshot of whole page
+     * @param basename filename (below screenshot base directory).
+     * @return location of screenshot.
+     */
+    public String fullPageScreenshot(String basename) {
+        return takeScreenshotWith(
+            basename,
+            name -> 
+                getSeleniumHelper()
+                    .fullPageScreenshot(
+                        getScreenshotBasename(name)
```

Comment:
Why is the extra `getScreenshotBasename` required here and not above?

Furthermore it seems a bit odd that both the basename is supplied to takeScreenshotWith, and all it does is pass it to the function. Seems easier if the caller would just place it inside the lambda (and then pass Supplier<String> instead of function)